### PR TITLE
[Chore] Improve test coverage (Programming Assessment)

### PR DESCRIPTION
**Pull request by derLalla**
### Description

This PR adds missing test cases for the programming assessment repo export dialog. Also, it fixes a bug, which was due to a always true value in an if statement.

### Steps for Testing

No changes to the actual code. Only test cases.

### Test Coverage

main/webapp/app/shared/util/download.util.ts: 13.33% -> 93.33%
main/webapp/app/exercises/programming/assess/repo-export/programming-assessment-repo-export-dialog.component.ts: 27.59% -> 87.93%

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/assess/repo-export/programming-assessment-repo-export-dialog.component.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ export class ProgrammingAssessmentRepoExportDialogComponent implements OnInit {
         }
         this.exportInProgress = true;
         // The participation ids take priority over the participant identifiers (student login or team names).
-        if (this.participationIdList) {
+        if (this.participationIdList.length) {
```
Comment (FuchsDominik) [inline src/main/webapp/app/exercises/programming/assess/repo-export/programming-assessment-repo-export-dialog.component.ts:85]: Are you sure `participationIdList` cannot be undefined?

Comment (krusche) [reply src/main/webapp/app/exercises/programming/assess/repo-export/programming-assessment-repo-export-dialog.component.ts:85]:
```suggestion
        if (this.participationIdList?.length) {
```

Comment (krusche) [reply src/main/webapp/app/exercises/programming/assess/repo-export/programming-assessment-repo-export-dialog.component.ts:85]: might be saver to use the question mark operator here

Comment (derLalla) [reply src/main/webapp/app/exercises/programming/assess/repo-export/programming-assessment-repo-export-dialog.component.ts:85]: oh no I am not. In the client test I couldn't assign undefined. But I will fix this.

Comment (derLalla) [reply src/main/webapp/app/exercises/programming/assess/repo-export/programming-assessment-repo-export-dialog.component.ts]: fixed in 6908cd85d14c0b02c33ca323da1fc2f7493e7862

File: src/test/javascript/spec/component/programming-assessment/programming-assessment-repo-export-dialog.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            .then(() => {
+                // Ignore console errors
+                console.error = () => false;
+                fixture = TestBed.createComponent(ProgrammingAssessmentRepoExportDialogComponent);
+                comp = fixture.componentInstance;
+                exerciseService = fixture.debugElement.injector.get(ExerciseService);
+                repoExportService = fixture.debugElement.injector.get(ProgrammingAssessmentRepoExportService);
+
+                // stubs
+                findExerciseId = stub(exerciseService, 'find').returns(of({ body: programmingExercise } as HttpResponse<Exercise>));
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/programming-assessment/programming-assessment-repo-export-dialog.component.spec.ts:69]: The `findExerciseId` variable doesn't seem to be used. (If you need the stub statement I'd just remove the variable assignment.)

File: src/test/javascript/spec/component/programming-assessment/programming-assessment-repo-export-dialog.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    it('test initialization', () => {
+        fixture.detectChanges();
+        expect(comp.exerciseId).to.be.equal(42);
+    });
+
+    it('Exerciseservice should find the correct programming exercise', fakeAsync(() => {
+        fixture.detectChanges();
+        expect(comp.exercise).to.be.equal(programmingExercise);
+    }));
```
Comment (FrankeLukas) [inline src/test/javascript/spec/component/programming-assessment/programming-assessment-repo-export-dialog.component.spec.ts:85]: Is it necessary to `fakeAsync` here (without `tick` etc.)?

**Review by krusche — APPROVED** [review]

**Pull request merged**